packages/server/graphql/mutations/helpers/convertContentStateToADF.ts

```diff
@@ -0,0 +1,247 @@
+import {ContentState, ContentBlock} from 'draft-js'
+import {Constants, getEntityRanges} from 'draft-js-utils'
+
+const {INLINE_STYLE, BLOCK_TYPE, ENTITY_TYPE} = Constants
+
+interface Mark {
+  type: string
+  attrs?: Record<string, any>
+}
+
+interface Text {
+  type: 'text'
+  text: string
+  marks?: Mark[]
+}
+
+interface Node {
+  type: string
+  attrs?: Record<string, any>
+  content: (Node | Text)[]
+}
+
+interface Doc {
+  version: 1
+  type: 'doc'
+  content: Node[]
+}
+
+class ContentStateToADFConverter {
+  private contentState: ContentState
+
+  private renderText = (text: string, marks?: Mark[]): Text => ({
+    type: 'text',
+    text,
+    ...(marks?.length ? {marks} : {})
+  })
+
+  private renderContent = (block: ContentBlock): Node[] => {
+    const text = block.getText()
+    const charMetaList = block.getCharacterList()
+    const entityPieces = getEntityRanges(text, charMetaList)
+    const content = entityPieces.flatMap(([entityKey, stylePieces]) => {
+      const contentPieces: Text[] = stylePieces
+        .map(([text, style]) => {
+          if (!text) {
+            return null
+          }
+
+          if (style.has(INLINE_STYLE.CODE)) {
+            return {
+              type: 'text',
+              text,
+              marks: [
+                {
+                  type: 'code'
+                }
+              ]
+            }
+          }
+          const marks: Mark[] = []
+          if (style.has(INLINE_STYLE.BOLD)) {
+            marks.push({
+              type: 'strong'
+            })
+          }
+          if (style.has(INLINE_STYLE.UNDERLINE)) {
+            marks.push({
+              type: 'underline'
+            })
+          }
+          if (style.has(INLINE_STYLE.ITALIC)) {
+            marks.push({
+              type: 'em'
+            })
+          }
+          if (style.has(INLINE_STYLE.STRIKETHROUGH)) {
+            marks.push({
+              type: 'strike'
+            })
+          }
+          return this.renderText(text, marks)
+        })
+        .filter((content) => content !== null)
+
+      const entity = entityKey ? this.contentState.getEntity(entityKey) : null
+      if (entity && entity.getType() === ENTITY_TYPE.LINK) {
+        const data = entity.getData()
+        const mark = {
+          type: 'link',
+          attrs: {
+            href: data.href || data.url || '',
+            ...(data.title ? {title: data.title} : {})
+          }
+        }
+        return contentPieces.map((piece) => ({
+          ...piece,
+          marks: [mark, ...(piece.marks ?? [])]
+        }))
+      }
+      /*
+      TODO: In order to show images or other embedded types, we need to upload those separately to Jira and then refer to them by id
```

Comment:
AOK to skip, our draftjs implementation doesn't support images, so we couldn't push em even if we wanted to